react-components/src/context/__test__/NavigationContext.test.js

```diff
@@ -0,0 +1,205 @@
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+import React from 'react';
+import { render, fireEvent, waitForElement } from '@testing-library/react';
+import { MockedProvider } from '@apollo/react-testing';
+import NavigationContextProvider, { useNavigationContext } from '../NavigationContext';
+
+describe('NavigationContext test', () => {
+    const ContextWrapper = () => {
+        const [
+            { view },
+            { showSignIn, showMyAccount, showForgotPassword, showChangePassword, showCreateAccount, showAccountCreated }
+        ] = useNavigationContext();
+
+        let child;
+        let content;
+
+        switch (view) {
+            case 'MENU':
+                child = <div data-testid="menu">Menu page</div>;
+                break;
+            case 'SIGN_IN':
+                child = <div data-testid="sign_in">Sign in page</div>;
+                break;
+            case 'MY_ACCOUNT':
+                child = <div data-testid="my_account">My account page</div>;
+                break;
+            case 'CHANGE_PASSWORD':
+                child = <div data-testid="change_password">Change password page</div>;
+                break;
+            case 'FORGOT_PASSWORD':
+                child = <div data-testid="forgot_password">Forgot password page</div>;
+                break;
+            case 'CREATE_ACCOUNT':
+                child = <div data-testid="create_account">Create account page</div>;
+                break;
+            case 'ACCOUNT_CREATED':
+                child = <div data-testid="account_created">Account created page</div>;
+                break;
+        }
+
+        content = (
+            <>
+                <div>{child}</div>
+                <button onClick={showSignIn}>Sign in</button>
+                <button onClick={showMyAccount}>My account</button>
+                <button onClick={showForgotPassword}>Forgot password</button>
+                <button onClick={showChangePassword}>Change password</button>
+                <button onClick={showCreateAccount}>Create account</button>
+                <button onClick={showAccountCreated}>Account created</button>
+            </>
+        );
+
+        return <div>{content}</div>;
+    };
+
+    it('shows sign in page', async () => {
+        const { getByText, getByTestId } = render(
+            <MockedProvider addTypename={false}>
+                <NavigationContextProvider>
+                    <ContextWrapper />
+                </NavigationContextProvider>
+            </MockedProvider>
+        );
+
+        expect(getByText('Sign in')).not.toBeUndefined();
```

Comment:
`expect(getByText('Sign in'))` should be sufficient, since the `getBy*` query throws an error if an element cannot be found. See https://testing-library.com/docs/dom-testing-library/api-queries#getby.

Reply:
True but `expect` is rarely used without a matcher.
It also fails linting.